go/pkg/filemetadata/filemetadata_test.go

```diff
@@ -64,7 +78,62 @@ func TestComputeFiles(t *testing.T) {
 	}
 }
 
+func TestComputeFilesWithXattr(t *testing.T) {
+	XattrDigestName = "google.digest.sha256"
+	XattrAccess = xattributeAccessorMock{}
+	tests := []struct {
+		name       string
+		contents   string
+		executable bool
+	}{
+		{
+			name:     "empty",
+			contents: "",
+		},
+		{
+			name:     "non-executable",
+			contents: "bla",
+		},
+		{
+			name:       "executable",
+			contents:   "foo",
+			executable: true,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			getXAttrMock = func(_ string, _ string) ([]byte, error) {
+				return []byte(tc.name), nil
+			}
+
+			before := time.Now().Truncate(time.Second)
+			filename, err := testutil.CreateFile(t, tc.executable, tc.contents)
```

Comment:
We can't even guarantee it's being run on a file system with extended attribute support.  That's what the Mock is for.  My workstation, including the /tmp folder, doesn't have extended attributes turned on, I had to do all integration testing out of google3.
By using the mock and providing a fixed string to return it also gives us verifiability that the Compute() function is actually using the attribute; otherwise the digest for the xattribute and for hashing the file would be identical and I can't think of a way to prove that two identical outputs were produced via two different code paths.

Reply:
> We can't even guarantee it's being run on a file system with extended attribute support. That's what the Mock is for. My workstation, including the /tmp folder, doesn't have extended attributes turned on, I had to do all integration testing out of google3.

Aha ok, makes sense.